src/lib.rs

```diff
@@ -832,12 +832,13 @@ fn encode_asn1_string(tag: u8, force_chars: bool, c: ASN1Class, s: &String)
     Ok(res)
 }
 
-fn encode_tag(c: ASN1Class, t: &BigUint) -> Vec<u8> {
+fn encode_tag(c: ASN1Class, t: &BigUint, primitive: bool) -> Vec<u8> {
     let cbyte = encode_class(c);
+    let pbyte = if primitive { 0 } else { 0b100000 };
 
     match t.to_u8() {
         Some(x) if x < 31 => {
-            vec![cbyte | x]
+            vec![cbyte | pbyte | x]
         }
         _ => {
             let mut res = encode_base127(t);
```

Comment:
This appears to mask off the ASN.1 class associated with this tag. Is there something about large tag values that means that the class *must* be universal? I would have expected a mask of `0b11011111` given the rest of your description.

Reply:
Not sure I follow. 

Class byte should be at most `0b11_00_00_00`.
Encoding form byte is `0b00_10_00_00`.